src/screens/History/MultiChainHistoryEtherspot.js

```diff
@@ -140,3 +157,63 @@ function useHistoryEvents(chain: Chain): Event[] {
     ...mappedCollectiblesHistory,
   ];
 }
+
+function useWalletEvents(chain: Chain): WalletEvent[] {
+  const walletEvents = useRootSelector((root) => root.walletEvents.data);
+  const activeAccount = useActiveAccount();
+
+  if (!activeAccount) return [];
+
+  const activeAccountId = getAccountId(activeAccount);
+  const accountChainWalletEvents = walletEvents?.[activeAccountId]?.[chain] ?? [];
+
+  // $FlowFixMe: does not cover WalletActivated and flow fails
+  return accountChainWalletEvents.map((event) => ({
+    ...event,
+    date: parseDate(event.date),
+  }));
+}
+
+function useEnsRegisteredEvent(chain: Chain): ?EnsNameRegisteredEvent {
```

Comment:
Consider using extracting this as memoized selector to avoid expensive computation on each render.